### PR TITLE
feat(offers): remove provider from offer

### PR DESCRIPTION
**Pull request by Phil91**
## Description

instead of having a separate property for the provider the providing company name is used

## Why

The provider is kind of duplicated, since the offer is linked to a company which represents the provider the name of the company is better

## Issue

Refs: #811

## Checklist

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed [IP checks](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-04#checking-libraries-using-the-eclipse-dash-license-tool) for added or updated 3rd party libraries
- [x] I have created and linked IP issues or requested their creation by a committer
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally
- [x] I have added tests that prove my changes work
- [x] I have checked that new and existing tests pass locally with my changes
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

**Comment by ybidois** [comment]
@ntruchsess @evegufy any timeline for reviewing this? We're very interested in cherry-picking it.

**Comment by ntruchsess** [comment]
> @ntruchsess @evegufy any timeline for reviewing this? We're very interested in cherry-picking it.
@ybidois : I've returned from vacation and doing the review right now. Looks as if the change needs some adjusments but that shouldn't take too long.

**Review by ntruchsess — DISMISSED** [review]

**Review by evegufy — CHANGES_REQUESTED** [review]
please take care of the codeql findings

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs [hunk 1]
```diff
@@ -24,31 +24,19 @@
 using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities;
 using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Entities;
 using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Enums;
-using System.Linq.Expressions;
 
 namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.DBAccess.Repositories;
 
 /// Implementation of <see cref="IOfferRepository"/> accessing database with EF Core.
-public class OfferRepository : IOfferRepository
+public class OfferRepository(PortalDbContext dbContext) : IOfferRepository
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferRepository.cs:34]: fixed

File: src/portalbackend/PortalBackend.DBAccess/Repositories/OfferSubscriptionsRepository.cs [hunk 2]
```diff
@@ -24,34 +24,22 @@
 using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities;
 using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Entities;
 using Org.Eclipse.TractusX.Portal.Backend.PortalBackend.PortalEntities.Enums;
-using System.Linq.Expressions;
 
 namespace Org.Eclipse.TractusX.Portal.Backend.PortalBackend.DBAccess.Repositories;
 
 /// Implementation of <see cref="IOfferSubscriptionsRepository"/> accessing database with EF Core.
-public class OfferSubscriptionsRepository : IOfferSubscriptionsRepository
+public class OfferSubscriptionsRepository(PortalDbContext dbContext) : IOfferSubscriptionsRepository
```
[the remark this answers is not in the document]
Comment (ntruchsess) [reply src/portalbackend/PortalBackend.DBAccess/Repositories/OfferSubscriptionsRepository.cs:34]: fixed

**Comment by ntruchsess** [comment]
> please take care of the codeql findings

both fixed

**Review by ntruchsess — APPROVED** [review]

**Review by evegufy — APPROVED** [review]

**Pull request merged**